src/Symfony/Component/Messenger/Command/FailedMessagesRetryCommand.php

```diff
@@ -198,7 +205,12 @@ private function runWorker(string $failureTransportName, ReceiverInterface $rece
                 throw new \RuntimeException(sprintf('The message with id "%s" could not decoded, it can only be shown or removed.', $this->getMessageId($envelope) ?? '?'));
             }
 
-            $shouldHandle = $shouldForce || 'retry' === $io->choice('Please select an action', ['retry', 'delete'], 'retry');
+            $this->forceExit = true;
+            try {
+                $shouldHandle = $shouldForce || 'retry' === $io->choice('Please select an action', ['retry', 'delete'], 'retry');
+            } finally {
+                $this->forceExit = false;
+            }
```

Comment:
With the `messenger:failed:retry` it's a little different, the `$io->choice()` method has an infinitive loop, so we need to exit if the command is waiting for an answer, otherwise we're stuck in the loop, which is the original issue that I tried to fix in #50787.